protobuf/transform.go

```diff
@@ -55,16 +56,86 @@ func (t *Transformer) Transform(p *scanner.Package) *Package {
 		pkg.Enums = append(pkg.Enums, enum)
 	}
 
+	names := buildNameSet(p)
+	for _, f := range p.Funcs {
+		rpc := t.transformFunc(pkg, f, names)
+		if rpc != nil {
+			pkg.RPCs = append(pkg.RPCs, rpc)
+		}
+	}
+
 	return pkg
 }
 
+func (t *Transformer) transformFunc(pkg *Package, f *scanner.Func, names nameSet) *RPC {
+	name := f.Name
+	if f.Receiver != nil {
+		n, ok := f.Receiver.(*scanner.Named)
+		if !ok {
+			report.Warn("invalid receiver type for func %s", f.Name)
+			return nil
+		}
+
+		name = fmt.Sprintf("%s_%s", n.Name, name)
+	}
+
+	rpc := &RPC{
+		Name:   name,
+		Input:  t.transformInputTypes(pkg, f.Input, names, name),
```

Comment:
It's here https://github.com/src-d/proteus/pull/24/files#diff-a889a2f970dd32c45043c224d33fe874R90 but i forgot about variadicity, so I'm pointing that out there